### PR TITLE
[infra-proxy-service] Update infra views policies action

### PR DESCRIPTION
**Pull request by vsingh-msys**
### :nut_and_bolt: Description: What code changed, and why?

- Updated `infra:infraServers:get` action for all get operations cookbooks, roles , orgs etc.

- Merged server org permissions with the top-level chef-server permissions. ie. `infra:infraServers:update`  
 
- Added an `authz-service` DB migration that adds `infra:infraServers:get` and `infra:infraServers:list` as an actions to both the Viewer and Editors roles.

**Note:** Pending with UI changes to hide the "Add chef server" button if the user lack the permission.

### :chains: Related Resources
https://github.com/chef/automate/issues/3519#issuecomment-621392424

### :+1: Definition of Done
- Infra service policy actions have been updated.
- Infra views content access for `viewer` & `editor` role.

### :athletic_shoe: How to Build and Test the Change

- `rebuild components/infra-proxy-service`
- `rebuild components/automate-gateway`
- `rebuild components/authz-service`

Steps to test:

- Login with `editor` or `viewer` users credential able to perform all views operation and permission error for adding/deleting operations.

### :white_check_mark: Checklist

- [x] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [x] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### :camera: Screenshots, if applicable
Signed-off-by: Vivek Singh <vivek.singh@msystechnologies.com>

### CODE REVIEW

File: components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql [hunk 1]
```diff
@@ -0,0 +1,18 @@
+
+-- append "infra:infraServers:list" & "infra:infraServers:get" with "infra:nodes:*" and "infra:nodeManagers:*"
```
Comment (bcmdarroch) [inline components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql:2]:
```suggestion
-- append "infra:infraServers:list" & "infra:infraServers:get" to the editor role's actions
```

File: components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+-- for the editor role
+
+UPDATE iam_roles
+    SET actions = '{infra:infraServers:list,infra:infraServers:get}' || actions
+    WHERE
+        id = 'editor' AND
+        NOT actions @> '{infra:infraServers:list,infra:infraServers:get}';
+
+
+-- append "infra:infraServers:list" & "infra:infraServers:get" with "infra:nodes:*" and "infra:nodeManagers:*"
```
Comment (bcmdarroch) [inline components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql:12]:
```suggestion
-- append "infra:infraServers:list" & "infra:infraServers:get" to the viewer role's actions
```

**Comment by vsingh-msys** [comment]
Thanks @bcmdarroch 

Regarding:
 - UI needs `app-authorized` to hide the button if the user lacks permission updated in issue TODO https://github.com/chef/automate/issues/3538.

 - Adding validation for the admin key, still looking for a solution, will try to catch in validation PR.

File: components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+        id = 'editor' AND
+        NOT actions @> '{infra:infraServers:list,infra:infraServers:get}';
+
+
+-- append "infra:infraServers:list" & "infra:infraServers:get" to the viewer role's actions
+UPDATE iam_roles
+    SET actions = '{infra:infraServers:list,infra:infraServers:get}' || actions
+    WHERE
+        id = 'viewer' AND
+        NOT actions @> '{infra:infraServers:list,infra:infraServers:get}';
```
Comment (msorens) [inline components/authz-service/storage/postgres/migration/sql/77_update_infra_service_policies.up.sql:15]: Could be more concise without loss of clarity:
```suggestion
UPDATE iam_roles
    SET actions = '{infra:infraServers:list,infra:infraServers:get}' || actions
    WHERE
        (id='editor' or id='viewer') AND
        NOT actions @> '{infra:infraServers:list,infra:infraServers:get}';
```

**Pull request merged**